### DISCUSSION TITLE
Nvidia Xavier 16 GB performance

### DISCUSSION THREAD

**Discussion by koh-osug** [openai/whisper#597] [Q&A]
I have bought a Nvidia Xavier 16 GB based computer. I could use a medium model and the transcription for a 4 s audio file took 8 s. Is this expected?
I could also not load the large model. I always get as output on the command line "Killed", it seems it cannot be loaded. I though 16 GB RAM are enough in the GPU?

**Comment by luigisaetta** [comment]
@koh-osug From what you write it seems that the code is NOT running on GPU. I have tested on GPU and an audio long 130 sec can be processed in only 10 sec (with an old P100). (see: https://github.com/luigisaetta/whisper-app)
The **first thing you should check** is that your Linux distro on the Xavier is correctly configured to use CUDA (the low level libraries provided by NVIDIA to run code on GPU)

A simple test: run python and execute this short script:

```
import torch
print(f"GPU is available: {torch.cuda.is_available()}"
```

it must return True... otherwise it is a problem of the underlying config. You should be able to find online docs on how to find Linux to correctly use CUDA on Xavier.

**Reply by koh-osug** [reply]
Thanks for the project. I guess this is not my problem. Nvidia provides special versions of Torch which I installed. I have found indication that using the `decode` method is faster than `transcribe`. Not sure if this is true.

**Comment by FurkanGozukara** [comment]
I have GTX 3060 12 gb and it can transcribe large model

you can watch my video and read the comments here :

**How Good is RTX 3060 for ML AI Deep Learning Tasks and Comparison With GTX 1050 Ti and i7 10700F CPU**>  https://www.youtube.com/watch?v=q8Q8CCDdSKo

**Reply by koh-osug** [reply]
The Nvidia is using a shared memory model, so the 16 GB are actually not completely available. I will check out your video.

**Comment by kaoh** [comment]
The Nvidia Xavier 16 GB based computer is from Seeed. The first transcription is slow because the network has to  be initialized. After sorting it this issues the speed is almost 1:1, i.e. 10s speech = 10s transcription using the medium model.
